apps/one-drive/src/inngest/functions/data-protection/sync-items.ts

```diff
@@ -0,0 +1,302 @@
+import { logger } from '@elba-security/logger';
+import type { DataProtectionObject, DataProtectionObjectPermission } from '@elba-security/sdk';
+import { Elba } from '@elba-security/sdk';
+import { eq } from 'drizzle-orm';
+import { NonRetriableError } from 'inngest';
+import { env } from '@/env';
+import { inngest } from '@/inngest/client';
+import { db } from '@/database/client';
+import { organisationsTable } from '@/database/schema';
+import { decrypt } from '@/common/crypto';
+import type { MicrosoftDriveItem } from '../../../connectors/share-point/items';
+import { getItems } from '../../../connectors/share-point/items';
+import type { MicrosoftDriveItemPermissions } from '../../../connectors/share-point/permissions';
+import { getItemPermissions } from '../../../connectors/share-point/permissions';
+
+export type ItemsWithPermisions = {
+  item: MicrosoftDriveItem;
+  permissions: MicrosoftDriveItemPermissions[];
+};
+
+export const parseItems = (
+  inputItems: MicrosoftDriveItem[]
+): {
+  folders: MicrosoftDriveItem[];
+  items: MicrosoftDriveItem[];
+} => {
+  const folders: MicrosoftDriveItem[] = [];
+  const items: MicrosoftDriveItem[] = [];
+
+  inputItems.forEach((item) => {
+    if (item.folder) {
+      folders.push(item);
+    } else {
+      items.push(item);
+    }
+  });
+
+  return {
+    folders,
+    items,
+  };
+};
+
+export const getCkunkedArray = <T>(array: T[], batchSize: number): T[][] => {
+  const chunks: T[][] = [];
+  for (let i = 0; i < array.length; i += batchSize) {
+    chunks.push(array.slice(i, i + Number(batchSize)));
+  }
+  return chunks;
+};
+
+export const formatPermissions = (
+  permission: MicrosoftDriveItemPermissions
+): DataProtectionObjectPermission[] | [] => {
+  if (permission.grantedToV2?.user) {
+    return [
+      {
+        id: permission.id,
+        type: 'user',
+        displayName: permission.grantedToV2.user.displayName,
+        userId: permission.grantedToV2.user.id,
+        email: permission.grantedToV2.user.email,
+        metadata: {
+          permissionId: permission.id,
+          roles: permission.roles,
+        },
+      },
+    ];
+  } else if (permission.grantedToIdentitiesV2) {
+    const formattedPermissions: DataProtectionObjectPermission[] = [];
+
+    for (const p of permission.grantedToIdentitiesV2) {
+      if (p.user) {
+        formattedPermissions.push({
+          id: `${permission.id}${env.ID_SEPARATOR}${p.user.id}`,
+          type: 'user',
+          displayName: p.user.displayName,
+          userId: p.user.id,
+          email: p.user.email,
+          metadata: {
+            permissionId: permission.id,
+            roles: permission.roles,
+          },
+        });
+      }
+    }
+
+    return formattedPermissions;

```

Comment:
```suggestion
    return permission.grantedToIdentitiesV2
        .filter(({ user }) => user)
        .map(permissionGrant => ({
          id: `${permission.id}${env.ID_SEPARATOR}${permissionGrant.user.id}`,
          type: 'user',
          displayName: permissionGrant.user.displayName,
          userId: permissionGrant.user.id,
          email: permissionGrant.user.email,
          metadata: {
            permissionId: permission.id,
            roles: permission.roles,
          }
        })
```
Also, I am not sure if we should add the `ID_SEPARATOR` in env variable. If we need to change it, we will have to handle metadata version anyway with code. To do so with env variable could be a lot trickier than without it.